compat/test/browser/suspense-hydration.test.js

```diff
@@ -139,6 +139,73 @@ describe('suspense hydration', () => {
 		});
 	});
 
+	it('Should hydrate a fragment with multiple children correctly', () => {
+		scratch.innerHTML = '<div>Hello</div><div>World!</div>';
+		clearLog();
+
+		const [Lazy, resolve] = createLazy();
+		hydrate(
+			<Suspense>
+				<Lazy />
+			</Suspense>,
+			scratch
+		);
+		rerender(); // Flush rerender queue to mimic what preact will really do
+		expect(scratch.innerHTML).to.equal('<div>Hello</div><div>World!</div>');
+		expect(getLog()).to.deep.equal([]);
+		clearLog();
+
+		return resolve(() => (
+			<>
+				<div>Hello</div>
+				<div>World!</div>
+			</>
+		)).then(() => {
+			rerender();
+			expect(scratch.innerHTML).to.equal('<div>Hello</div><div>World!</div>');
+			expect(getLog()).to.deep.equal([]);
+
+			clearLog();
+		});
+	});
+
+	it('Should hydrate a fragment with multiple children and an adjacent node correctly', () => {
```

Comment:
React seems to also fail on this

```suggestion
    // This is an expected fail case, we should encourage folks to not have adjacent
    // nodes of the same type, or return a single wrapping dom node from a lazy boundary.
	it.skip('Should hydrate a fragment with multiple children and an adjacent node correctly', () => {
```